## Results

### Data augmentation via a two-letter degenerate amino acid alphabet increases the effective size of reference protein sequence databases by an order of magnitude

**Key: Reduced alphabets increase the effective size of the reference amino acid database.**
Fewer than 1% of the at least 10 million animals have 100 or more annotated genes in the UniProtKB/SwissProt (2022_01, 23-Feb-2022) database, when most animals species have 1000s of predicted genes (Figure 1a).
Thus, as sequence annotation is traditionally performed by raw sequence similarity to previously observed data, over 99% of Metazoans have little chance of having sequence similarity to existing data, and it is very difficult to assign any annotation.
In image-based machine learning, reference datasets are often expanded using data augmentation techniques such as flipping, translating, cropping, or rotating the original image to detect comparable features in a new context @doi:10.1109/IIPHDW.2018.8388338.
However, in the biological sequence space, data augmentation is challenging as not many schemas exist.
Recent work @https://doi.org/10.1101/2021.02.18.431877 investigated protein data augmentation using the replacement of individual amino acids, shuffling, reversion, and subsampling subsequences.
Reduced amino acid alphabets accomplish the task of data augmentation by implicitly encoding many protein synonyms of similar biochemical structures, thus expanding the adjacent possible protein sequences.
By expanding the reference database with related sequences, reduced alphabets help algorithms identify adjacently congruent sequences, that may not have exactly the same amino acid sequence but similar biochemical properties.
The full extent of k-mers compatible with Eukaryotic life is not yet known but can be implied by reduced amino acid alphabets.

- Compressing to a 2-letter alphabet now creates 1e-16 k-mers in the adjacent possible of k-mer protein sequences potentially compatible with life
    - Using a reduced alphabet increases the theoretical sensitivity of sequence matches by an order of magnitude
    - **Key: For species that previously had zero matches using precision sequence similarity algorithms such as BLAST or HMMER, reduced amino acid alphabets enable at least a starting point for sequence annotation**
- 
- We suggest that the adjacent possible of protein sequences compatible with eukaryotic life can be approximated using reduced amino acid alphabets
- If we assume that the hydrophobic and polar patterns encoded by protein k-mer words are indicative of binary sequences compatible with life, we can imply many more protein sequences than were initially observed
- 
- A very strict form of sequence annotation is orthology assignment, which not only specifies the sequence similarity but also requires that sequences originated from a common ancestor
    - While the reduced alphabets do not necessarily improve orthology matches (unknown) but they can expand the world of possible matches, making studying the 99.9% of species without reference genomes much more accessible
    - Applying the HP k-mers to the Quest for Orthologs data …

> Release 2022_01 of 23-Feb-2022 of UniProtKB/Swiss-Prot contains 566,996 sequence entries, curated from 284,170 unique references and comprising 204,698,499 amino acids.

The upper bound of k-mers is ~2e8, due to the total length of observed number of sequences. But of the 20^24,

Observed 24-mer protein words:

2e8/20^24 = 2/2 * (1e8/10e24) = (1e8/1e25) = 1e-17

Observed 24-mer hp words:

2e8/2e24 = 2/2 * 1e8/1e24 = 1e-16

(I think this next section is a better fit in methods)

For the hydrophobic-polar alphabet, the median replacement rate is 10, and thus each HP k-mer represents ~10^k underlying protein sequences, and this is 10^k more underlying protein sequences than that could have been encoded in the original database. Here, we use k=24, and 10^24 increase in encoded protein sequence

- 2^24 = 16,777,216 possible 24-long HP-mers
- E.g. hhhhhhhhhhhhpppppppppppp
- H = 10 possibilities
- P = 10 possibilities
- This sequence represents 10^24 possible amino acids
- 1,000,000,000,000,000,000,000,000
- 1 septillion possible 24-mer peptide sequences
- If using a protein alphabet of length k=24: 20^24 peptide sequences
- How does this compare when a database is involved?
- For a database size N, the upper bound on number of possible k-mers is N-k.
- However, when the k-mer size k << N, for example, with k=24 and N=204,698,499 (the latest Uniprot release), we can approximate this as N.
- The number of observed amino acid words, 20^k, is limited by the database size.
- With the latest Uniprot release, we can compute the breaking point where all possible 20-letter amino acid words are present: k=6 → 20^6 < 204,698,499
    - However, with k >=7, then we begin to have unobserved 20^-k-mer in the database
    - Are these sequences incompatible with life or have they simply not been observed/

Note that this only occurs with very degenerate alphabet sizes. The Dayhoff alphabet, with 6 categories, does not have this property. The breaking point for amino acid representations that create more sequences than previously shown is sqrt(20).

The size factor of how much a reduced alphabet increases the effective size of a database depends on the size of the reduced alphabet. Given the original 20-letter amino acid alphabet, a *reduced* amino acid alphabet of size |∑| will have a degeneracy rate of 20/|∑|. Thus, for any k-mer with k>1 in the reduced alphabet, it represents an underlying ((20/|∑|)/|∑|)^k = 20^k/|∑|^{2k} possible k-mers in the original amino acid alphabet. Given

**Key: While a database size is fixed, the effective size using a degenerate alphabet is orders of magnitude larger, encapsulating putative amino acid sequences that may still encode proteins of similar structure and function.**

For any k-mer size of k > 1, the point at which this ratio becomes greater than 1, meaning where a reduced alphabet k-mer encodes more sequences than its underlying protein sequence, on average


### Benchmarking: Reduced amino acid alphabets sensitively detect functional sequence categories across evolutionary time

- Paragraph 1: Statistical basis of homolog detection and sanity checks with mouse+human
    - **Key Sentence: Shared reduced amino acid k-mers reproduce homologous gene assignments between mouse and human**
    - (gather approach) As a sanity check, we first used a highly curated dataset of human and mouse homologs.
    - We used a greedy algorithm to find matches, implemented in “sourmash gather,” where the best hit with the most k-mers is found, and then removed from the search database.
        - Using an amino acid k-mer size of 24 and the hydrophobic-polar (HP) alphabet, we used “sourmash gather” to identify best-hit matches across all mouse and human protein sequence
        - For the MHC-containing chromosomes of chr6 for human and chr17 for mouse, 1317 pairs of known mouse-human homologs shared at least 1 24-mer.
            - The median overlap between two non-homologous gene was 1 k-mer, and thus we used a threshold of at least 2 k-mers overlapping.
        - Probability of a single k-mer overlap in human + mouse: 0.2171655826331076
            - Two k-mers overlapping: 0.04716089028037708
        - We used a greedy approach, finding the single best match in human, from a gene in mouse, and vice versa
        - 799 mouse-human gene pairs overlapped with 2 or more k-mers, not labeled as orthologous by the JAX lab informatics homologene mapping.
            - Of these, ___% were related gene families, such as heat shock proteins, Hspa1l in mouse matching to HSPA1B in human
            - Or mouse olfactory receptor 753, pseudogene 1(*Olfr753-ps1*), a polymorphic pseudogene and Olfactory Receptor Family 2 Subfamily H Member 2 (OR2H2) in human.
        - With a threshold of 2 or more shared subsampled k-mers, across human and mouse, we had very high recall:
            - precision: 0.6032770605759682
            - recall: 0.9225512528473804
            - fscore: 0.7295106574602221
            - Low precision is acceptable here as we are interested in sensitive methods that detect weak signals, and are willing to see a few more false positives with fewer false negatives.
            - The precision is low due to non-orthologous gene overlap, though genes may be in the same family
            - Using sensitive methods, we detecting more similar sequences than more specific methods


### Cross-phylum functional homologs identified between human *BCL2* and *Caenorhabditis elegans* Ced-9 gene

- Weighted similarity was obtained by dividing the cosine similarity between k-mer signatures by the probability of overlap between two proteins, using the underyling distribution of k-mers in each species as the null model.

### Cross-domain functional homologs identified between human *CD47* and *Borrelia* P83

### Functional homologs identified across all domains of life

- Reciprocal best hits found on quest for orthologs data
- Ideal ksize and protein sequence abstraction identified for bins of millions of years of divergence
